clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1657,6 +1658,87 @@ class SyclKernelDeclCreator : public SyclKernelFieldHandler {
   using SyclKernelFieldHandler::leaveStruct;
 };
 
+class SyclKernelNumArgsChecker : public SyclKernelFieldHandler {
+  SourceLocation KernelLoc;
+  unsigned NumOfParams = 0;
+
+  void addParam() { NumOfParams++; }
+
+  bool handleSpecialType(QualType FieldTy) {
+    const auto *RecordDecl = FieldTy->getAsCXXRecordDecl();
```

Comment:
Just a minor nit, I should have asked earlier, sorry.
Can the LHS be more specific here?   Could we do:
const CXXRecordDecl *
here?

Reply:
Ok, done